File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        strr = sorted(strs)
        minn = min(len(ele) for ele in strr)
        common = ""
        for i in range(minn):
            if strr[0][i]==strr[len(strr)-1][i]:
                common+=strr[0][i]
            else:
                break
        return common
      
        

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix = strs[0];
        for(string word: strs){
            for(int i = 0; i < word.size(); i++){
                if (i < prefix.size() && prefix[i] != word[i]){
                    prefix = prefix.substr(0,i);
                }
            
            }
            prefix =  prefix.substr(0,min(prefix.size(),word.size()));
        }
        return prefix;
    }
};